### PR TITLE
[Ada] Adding Ada support for client code generator

### PR DESCRIPTION
**Pull request by stcarrez**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming langauge.

### Description of the PR

This pull request adds Ada support for the a REST client code generator in Ada to solve #6601 

1. Added AbstractAdaCodegen class as the root Java class for Ada code generators
2. Added AdaCodegen class that inherits from AbstractAdaCodegen for the implementation of the code generator. The code generator only supports the client side. The server side will be provided as part of a next pull request (it will be made in the same class). I've followed several other code generators for the implementation.
3. Added several mustache templates in the resources/Ada directory for the code generator
4/ Updated the META-INF config file to add the new Ada generator class

The pull request does not contain the bin/ada-petstore.sh yet, nor the petstore samples files.

### CODE REVIEW

**Comment by wing328** [comment]
> Added AbstractAdaCodegen class as the root Java class for Ada code generators

👍 👍 👍

**Comment by wing328** [comment]
@stcarrez there are some minor docstring errors reported by the CIs:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.10.4:jar (attach-javadocs) on project swagger-codegen: MavenReportException: Error while generating Javadoc:
[ERROR] Exit code: 1 - /root/src/github.com/swagger-api/swagger-codegen/modules/swagger-codegen/src/main/java/io/swagger/codegen/InlineModelResolver.java:405: warning: no description for @return
[ERROR] * @return
[ERROR] ^
[ERROR] /root/src/github.com/swagger-api/swagger-codegen/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractAdaCodegen.java:96: error: unknown tag: brief
[ERROR] * @brief Turn a parameter name, operation name into an Ada identifier.
[ERROR] ^
[ERROR] /root/src/github.com/swagger-api/swagger-codegen/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractAdaCodegen.java:102: error: bad use of '>'
[ERROR] * body              -> P_Body
[ERROR] ^
[ERROR] /root/src/github.com/swagger-api/swagger-codegen/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractAdaCodegen.java:103: error: bad use of '>'
[ERROR] * petId             -> Pet_Id
[ERROR] ^
[ERROR] /root/src/github.com/swagger-api/swagger-codegen/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractAdaCodegen.java:104: error: bad use of '>'
[ERROR] * updatePetWithForm -> Update_Pet_With_Form
[ERROR] ^
[ERROR] /root/src/github.com/swagger-api/swagger-codegen/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AdaCodegen.java:212: error: unknown tag: brief
[ERROR] * @brief Post process the media types (produces and consumes) for Ada code generator.
[ERROR] ^
[ERROR] /root/src/github.com/swagger-api/swagger-codegen/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/StaticHtmlGenerator.java:80: warning - Tag @link: can't find processCompiler(Compiler) in io.swagger.codegen.languages.StaticHtmlGenerator
[ERROR] /root/src/github.com/swagger-api/swagger-codegen/modules/swagger-codegen/src/main/java/io/swagger/codegen/utils/Markdown.java:21: warning: nested tag not allowed: <code>
[ERROR] * @param markdown text with Markdown styles. If <code>null<code>, </code>""</code> is returned.
[ERROR] ^
[ERROR] 
```

Please take a look when you've time.

**Comment by stcarrez** [comment]
I've fixed javadoc issues and verified it was correct by running:
`mvn verify`

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AdaCodegen.java [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+    }
+
+    @Override
+    public String getName() {
+        return "Ada";
+    }
+
+    @Override
+    public String getHelp() {
+        return "Generates an Ada client implementation";
```
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AdaCodegen.java]: I'll add the word "beta" as we roll it out to the community and we can remove it later as the generator matures.

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AdaCodegen.java [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+    }
+
+    @Override
+    public CodegenType getTag() {
+        return CodegenType.CONFIG;
+    }
+
+    @Override
+    public String getName() {
+        return "Ada";
```
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AdaCodegen.java]: I'll rename it to "ada" as the convention is [kebab-case](https://en.wikipedia.org/wiki/Letter_case#Special_case_styles)

Ref: https://github.com/swagger-api/swagger-codegen/wiki/How-to-add-a-generator-for-a-new-language-or-framework#checklist-a-list-of-items-we-look-for-in-the-new-generator

Comment (stcarrez) [reply modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AdaCodegen.java]: Ok. For your info, Ada is the name of a person: Ada Lovelace, which is why it has a capital letter.
I'll do the changes this evening. Thanks for taking time to review my pull request.

Looking at other generators, CsharpDotNet2 is not compliant then as it uses:

CsharpDotNet2ClientCodegen.java-        return "CsharpDotNet2";

Comment (wing328) [reply modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AdaCodegen.java]: > Ok. For your info, Ada is the name of a person: Ada Lovelace, which is why it has a capital letter.

Yup, I read that from https://en.wikipedia.org/wiki/Ada_(programming_language)

> CsharpDotNet2

Yes, I planned to change that (as reminded by this PR). There are only 2 that needs to be updated to conform to kebab-case.

Comment (wing328) [reply modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AdaCodegen.java]: FYI. PRs for renaming the other 2 generators:
https://github.com/swagger-api/swagger-codegen/pull/6612
https://github.com/swagger-api/swagger-codegen/pull/6611

**Comment by wing328** [comment]
@stcarrez thanks for the new generator, which has been merged into master. I've also updated README to list you as the template creator and sent out the following tweet to promote the new generator:

https://twitter.com/wing328/status/915426107901435904

**Pull request merged**